controllers/ovndbcluster_controller.go

```diff
@@ -579,8 +590,101 @@ func (r *OVNDBClusterReconciler) reconcileServices(
 			if err != nil {
 				return ctrl.Result{}, err
 			}
+			// Delete DNS thingy
+			dnsRecord := &infranetworkv1.DNSData{}
+			dnsName := "dns-" + serviceName + fmt.Sprintf("-%d", i)
+			dnsNamespace := helper.GetBeforeObject().GetNamespace()
+			Log.Info(fmt.Sprintf("Name: %v, Namespace: %v", dnsName, dnsNamespace))
+			test_err := helper.GetClient().Get(ctx, types.NamespacedName{Name: dnsName, Namespace: dnsNamespace}, dnsRecord)
+			if test_err != nil {
+				Log.Info(fmt.Sprintf("Error: %v", test_err))
+			}
+			err := helper.GetClient().Delete(ctx, dnsRecord)
+			if err != nil {
+				Log.Info(fmt.Sprintf("Error: %v", err))
+			}
+
+		}
+	}
+
+	Log.Info("DNS Starts here")
+
+	serviceLabels = map[string]string{
+		common.AppSelector: serviceName,
+	}
+	podList, err = ovndbcluster.OVNDBPods(ctx, instance, helper, serviceLabels)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	for i, ovnPod := range podList.Items {
+		if i >= int(*(instance.Spec.Replicas)) {
+			Log.Info(fmt.Sprintf("In iteration %v is OOO, exit", i))
+			break
+		}
+		var dnsName string
+		var dnsIP string
+		var hostnames []string
+		dnsName = "dns-" + ovnPod.Name
+		// Get Hostname
+		svc, _ := service.GetServiceWithName(
+			ctx,
+			helper,
+			ovnPod.Name,
+			ovnPod.Namespace,
+		)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		hostname := svc.ObjectMeta.Annotations[infranetworkv1.AnnotationHostnameKey]
+		hostnames = append(hostnames, hostname)
+
+		// Get IP
+		net_stat, _ := nad.GetNetworkStatusFromAnnotation(ovnPod.Annotations)
+		for _, v := range net_stat {
+			if v.Interface == instance.Spec.NetworkAttachment {
+				dnsIP = v.IPs[0]
+			}
+		}
+
+		// Create DNSRecord
+		var DNSRecords []infranetworkv1.DNSHost
+		// ovsdbserver-sb-x entry
+		DNSRecord := infranetworkv1.DNSHost{}
+		DNSRecord.IP = dnsIP
+		DNSRecord.Hostnames = hostnames
+		// ovsdbserver-sb entry
+		headless_dns_hostname := ovndbcluster.ServiceNameSB + "." + instance.Namespace + ".svc"
+		if instance.Spec.DBType == v1beta1.NBDBType {
+			headless_dns_hostname = ovndbcluster.ServiceNameNB + "." + instance.Namespace + ".svc"
 		}
+		DNSRecordCname := infranetworkv1.DNSHost{}
+		DNSRecordCname.IP = dnsIP
+		DNSRecordCname.Hostnames = append(DNSRecordCname.Hostnames, headless_dns_hostname)
+		DNSRecords = append(DNSRecords, DNSRecord)
+		DNSRecords = append(DNSRecords, DNSRecordCname)
+
+		// Create DNSData
+		DNSData := &infranetworkv1.DNSData{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      dnsName,
+				Namespace: ovnPod.Namespace,
+			},
+		}
+		_, err := controllerutil.CreateOrPatch(ctx, helper.GetClient(), DNSData, func() error {
+			DNSData.Spec.Hosts = DNSRecords
+			DNSData.Spec.DNSDataLabelSelectorValue = "dnsdata" //This is the default
```

Comment:
https://github.com/openstack-k8s-operators/infra-operator/blob/main/config/crd/bases/network.openstack.org_dnsdata.yaml#L47-L51

Since using default works and I (at the moment) I don't see any benefit from changing value, will delete this line